### DISCUSSION TITLE
Config for FLSUN SuperRacer

### DISCUSSION THREAD

**Discussion by systmworks** [Desuuuu/klipper#73] [Q&A] [answered]
Hello,
Thank you for your efforts on this project. 
I will try and provide as much info as I can think of, to help support my query. 

I am trying to get this working on an FLSUN SuperRacer delta printer.  
https://flsun3d.com/products/super-racer-sr 

The supplied screen is a DWIN DMG48320C035

The printer has a clone SKR 1.3 board  (LPC1768 / smoothieware)
https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3/tree/master/BTT%20SKR%20V1.3/hardware 

The screen is connected (4-wires) to the "TFT" port just above the "AUX-1" port in the lower center of this image:
https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3/blob/master/BTT%20SKR%20V1.3/Wiring%20diagram/SKR%20V1.3%20Wiring%20diagram.jpg 

I have followed the instructions below for adding the custom Klipper build and re-creating the printer firmware.
https://3dprintbeginner.com/how-to-install-klipper-on-creality-ender-6/#Merge_Custom_Klipper_build 

During the make menuconfig I chose "UART1 (on P0.16 / P0.15)"  
"UART0" and "UART3 (on P4.29 / P4.28)" were also available. 
I have not tried UART0 but when I chose UART3 Klipper could not connect to the printer - but can connect on UART1.

I flashed the screen with the DWIN_SET  folder from the link below - which seemed successful (10 "download" files and "SD Card Process... END !" message)
https://github.com/Desuuuu/DGUS-reloaded-Klipper/releases/  

However when powering the printer on the screen remains blank - no Klipper logo. 
Even if the screen lost connection with the mainboard, it should still display the Klipper logo from that firmware?

I added this to my printer.cfg file:
[t5uid1]
firmware: dgus_reloaded

Is there any way I can confirm, from Putty or the Klipper web GUI,  that the Desuuuu version of Klipper is running?
I saw the warning that the Klipper updates may over-write it?

I appreciate any suggestions.    There are quite a few other people in the SR Facebook group who have switched from Marlin to Klipper,  or considering it,   and retaining some screen functionality would be great.

Thank you,
Darren

**Comment by ihrapsa** [comment]
Looking it up online, that DWIN display is T5L based not T5UID1. Probably that's why it didn't work. I'm also looking forward to a way to easily add suport for different UART TFTs in Klipper (T5UIC1, T5L1). Desuuuu's been working lately on a sort of UART passthrough that would ease the process of porting theses kind of displays in Klipper.

**Comment by Desuuuu** [comment, marked as answer]
The touchscreen firmware only works with 480x272 T5UID1 screens. This touchscreen is a T5L one (from what I can tell), with a different resolution as well. It will not work.

As to determining whether you're running this fork or not: if the `[t5uid1]` section in your configuration doesn't just make Klippy crash on startup, you are running this fork.

**Reply by systmworks** [reply]
Thank you for the prompt replies.  Ahh thats a shame its not the correct model screen to work  :( 
I have the original FLSUN screen firmware I can flash back on there - but they dont release the project file for it so we cant make any changes to it (for use on Marlin). 
Oh well - I will keep my eyes open - hopefully one day more LCD models are supported by Klipper.

**Reply by ratomas108** [reply]
Hello,
try this:
https://github.com/Desuuuu/DGUS-reloaded-Klipper/discussions/33
